<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>bootcamp-intro-R_2023-03.knit</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Nevada INBRE DSCore</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Data Science Initiative
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DSI_welcome.html">Welcome</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Funding Opportunities
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="resource_service_awards.html">Scientific Core Service Awards</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Training Opportunities
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Upcoming</li>
    <li>
      <a href="workshop-research-computing-2023-11.html">2023 Fall: Research Computing</a>
    </li>
    <li>
      <a href="underConstruction.html">2024 Winter: Data Viz with R</a>
    </li>
    <li>
      <a href="underConstruction.html">2024 Spring Break: TBD</a>
    </li>
    <li class="dropdown-header">Previous</li>
    <li>
      <a href="bootcamp-r-series.html">2021 Summer: R for Data Science at UNR</a>
    </li>
    <li>
      <a href="bootcamp-supervised-ml-2022-10.html">2022 Fall: Python for ML at UNR</a>
    </li>
    <li>
      <a href="bootcamp-supervised-ml-2023-01.html">2023 Winter: Python for ML at UNR</a>
    </li>
    <li>
      <a href="NSC_ResearchDay_2023-03.html">2023 Spring: Research Day at NSC</a>
    </li>
    <li>
      <a href="bootcamp-intro-R_2023-03.html">2023 Spring: Programming with R at UNR</a>
    </li>
    <li>
      <a href="CSN_phageGenomics_2023-03.html">2023 Spring: Phages Genomics at CSN</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="jupyter-notebook-setup.html">Jupyter Notebook</a>
    </li>
    <li>
      <a href="NSC_IntrotoR.html">R</a>
    </li>
    <li class="dropdown-header">Julia</li>
    <li class="dropdown-header">Python</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">
    <span class="fa fa-address-book"></span>
     
    About
  </a>
</li>
<li>
  <a href="https://github.com/NevadaINBRE">
    <span class="fa fa-code-branch"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<style><!--Page specific style-->
  h1 {color: #000}
  h2 {color: #111}
  h3 {color: #222; border-bottom: 1px solid #ccc; width: 75%;}
  h4 {color: #666}
  .top {color: #ccc; display: inline-block; padding: 25px 0px 5px 0px;}
</style>
<p><img src="media/img/banner_Intro_R.png" style="width: 800px; float: top; margin: 20px 20px 20px 20px;"></p>
<p><span style="font-size: 14pt;">Interested in getting started with R
programming, but don’t know anything about programming or data analysis
… then come join us for our two-day bootcamp. </span><br></p>
<p>This training will be focused around <strong>hands-on learning
activities</strong> and you are welcome to submit your own data prior to
the workshop - if you are lucky it will be chosen as an example data
set.</p>
<p>You will be introduced to the <a href="https://www.r-project.org/">R
programming language</a> and to a popular editor
<a href="https://posit.co/downloads/">Posit, formerly known as
RStudio</a> while learning a little about Data Science. Example data
sets will be used to practice data manipulation, organization, and
graphical exploration. You are also encouraged to bring your own data
and data challenges. In addition, major concepts of scientific research
such as reproducibility will be addressed. Participants will gain the
skills necessary to manage and organize data, run basic analyses, and
generate beautiful, high-quality figures in R.</p>
<p><a href="https://www.canva.com/design/DAFakeR09b4/view">Here</a> is
the flyer, feel free to post it at your bulletin boards.<br></p>
<div id="bootcamp-overview" class="section level2">
<h2>Bootcamp Overview</h2>
<p>R is a complete, flexible, and open source system for statistical
analysis and graphics, and has become a tool of choice for many life
scientists.</p>
<div id="day-1-r-crash-course-with-introduction-to-data-analytics"
class="section level3">
<h3>Day 1: R Crash Course with Introduction to Data Analytics</h3>
<p><img src="media/img/data_point.png" style="width: 230px; float: right; margin: 5px 10px 10px 20px;"/>Day
1 will be an introduction to programming with R and teach you how to use
Rstudio. R basics will be covered, such as using the command line as a
calculator, storing values in a variable, <strong>basic data structures
(vector, matrix, list, data frame, etc.)</strong>, and data types. We
will show how to <strong>read-in basic data sets</strong> and access
rows/columns, and summarize data with statistics and plots. Then we will
move on to <strong>scripting in R, including control flow (choices and
loops), writing functions, and installing/loading libraries</strong>
(and the differences between CRAN, GitHub, and Bioconductor
packages).</p>
</div>
<div id="day-2-tidy-r-figures-and-reproducibility"
class="section level3">
<h3>Day 2: Tidy R, Figures, and Reproducibility</h3>
<p>We will discuss commonly seen coding problems and introduce what it
means to <strong>conduct reproducible data science</strong>. During the
second day, we will go through an example of <strong>setting up a new
project folder, loading “messy” data, organizing scripts, creating high
quality plots, and creating output files</strong>. In reference to the
messy data, we will go over what “tidy” data is, and how to use
<code>tidyverse</code> libraries. Then will go over some
<strong>principles of the visual display of quantitative data</strong>,
and go more into plotting in R with <code>base</code> and
<code>ggplot2</code>. <br></p>
</div>
</div>
<div id="schedule" class="section level2">
<h2>Schedule</h2>
<table class="tg">
<thead>
<tr>
<th>
Outline
</th>
<th>
Date and Time
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
R Crash Course with Introduction to Data Analytics
</td>
<!--(#intro1) before td to make a link-->
<td>
March 20, 2023, 9 - 4pm
</td>
</tr>
<tr>
<td>
Tidy R, Figures, and Reproducibility
</td>
<!--(#intro2)-->
<td>
March 21, 2023, 9 - 4pm
</td>
</tr>
</tbody>
</table>
<p><span style="font-size: 10pt;"><strong>noon - 1pm: lunch on your
own</strong></span> <br></p>
</div>
<div id="things-to-know" class="section level2">
<h2>Things to Know</h2>
<div id="audience-and-prerequisites" class="section level3">
<h3>Audience and Prerequisites</h3>
<p><strong>No prior knowledge about data analytics or coding experience
are necessary!</strong></p>
<p>This bootcamp is intended to beginners wanting to become familiar
with the R syntax, environment, and the most common commands to start
using R to explore, interpret, and present their data.</p>
<p>This <strong>is NOT a training on statistics</strong> but rather a
training on how to use R to perform different tasks.</p>
<p>You will need your own laptop and must install
<a href="https://www.r-project.org/">R language</a> and
<a href="https://posit.co/downloads/">Posit (aka RStudio)</a> prior to
the training; follow the instructions on their websites. We recommend to
first download R and then Posit. If you don’t have a laptop, please let
us know and we will provide a loaner laptop.</p>
<p>You are encouraged to go through some of the R documentation: <a
href="https://www.r-project.org/"
class="uri">https://www.r-project.org/</a>.</p>
</div>
<div id="application-and-registration-fee" class="section level3">
<h3>Application and Registration Fee</h3>
<div style="text-align: center">
<span style="font-size: 16pt;"><strong>The registration fee for this
workshop is $180.</strong> </span>
</div>
<div style="text-align: center">
<span style="font-size: 16pt;"><strong>Limited registration waivers are
available, everyone is welcome to apply!</strong> </span>
</div>
<p><br></p>
<p>We have a limited number of registration fee waivers available, but
we also encourage your professors to support your participation.
Application is open until <strong>March 13th, 2023 </strong>, but will
close as soon as spots are filled with confirmed participants.</p>
<p>Application is required to be considered to participate. Students and
faculty from TMCC, WNC, SNU, and UNR interested in R are encouraged to
apply. This is a two day workshop and participants must attend both days
to be considered for a registration waiver.</p>
<p><strong>For Students and Postdocs</strong>, application requires:</p>
<ul>
<li>Complete online application form</li>
<li>Indicate 1 academic reference, and provide their contact information
in the application form. The application will automatically send them a
link to the brief reference form to fill out. All references need to be
submitted directly by the author. Your references will not receive the
link to the reference form until your application is fully
submitted.</li>
</ul>
<p>If you are interested in applying for a Registration Waiver addition
materials are required:</p>
<ul>
<li>Copy of unofficial transcript (not for postdocs)</li>
<li>Academic references (need to provide an account number/worktag)</li>
<li>Personal statements not to exceed one page addressing the following
<ul>
<li>Academic background and research interests</li>
<li>Why you are interested in this bootcamp and what you hope to gained
from this training</li>
<li>How will this training affect your academic/research/professional
career</li>
<li>Anything else you want to share</li>
</ul></li>
</ul>
<div style="text-align: center">
<span
style="font-size: 14pt;"><a href="https://nvideaoffice.formstack.com/forms/programmingwithr">Click
here to be taken to the student/postodoc application form</a></span>
</div>
<p>You will receive an email to confirm participation, upon reception of
the confirmation email, participants will be asked to confirm their
attendance within 2 days.</p>
<p><strong>For Faculty and Staff</strong>, application requires:</p>
<ul>
<li>Complete online application form</li>
<li>Limited Registration Waivers are available, must indicate why one
cannot self-support their attendance</li>
</ul>
<div style="text-align: center">
<span
style="font-size: 14pt;"><a href="https://nvideaoffice.formstack.com/forms/programmingwithr_faculty">Click
here to be taken to the facutly/staff application form</a></span>
</div>
</div>
<div id="additional-information" class="section level3">
<h3>Additional Information</h3>
<ul>
<li>Applications from TMCC, WNC, and SNU will receive priority</li>
<li>No-show policy: the associated lab/PI will be charged the
registration fee even when a waiver was granted. In other words, if you
don’t intend to actually be there, don’t register and take a seat from
another student</li>
<li>Attendance is required in-person</li>
<li>Participants attending the entire bootcamp may request a certificate
of completion.</li>
</ul>
<p>Coordinator: Juli Petereit</p>
<p>Instructors: Nicole Schrad and Juli Petereit</p>
<!-- --- -->
<!-- ## Schedule {#schedule} -->
<!-- <table class="tg"> -->
<!-- <thead> -->
<!--   <tr> -->
<!--     <th>Workshop</th> -->
<!--     <th>Date and Time</th> -->
<!--   </tr> -->
<!-- </thead> -->
<!-- <tbody> -->
<!--   <tr> -->
<!--     <td>[Introduction to R Part 1](#intro1)</td> -->
<!--     <td>March 20, 2023, 9-12</td> -->
<!--   </tr> -->
<!--   <tr> -->
<!--     <td>[Introduction to R Part 2](#intro2)</td> -->
<!--     <td>March 20, 2023, 1-4pm</td> -->
<!--   </tr> -->
<!--   <tr> -->
<!--     <td>[Reproducible Data Science and Tidy R](#reproduce)</td> -->
<!--     <td>March 21, 2023, 9-12pm</td> -->
<!--   </tr> -->
<!--   <tr> -->
<!--     <td>[Producing Clean Documents, Graphs, and Tables](#tidy)</td> -->
<!--     <td>March 21,2023 1-4pm</td> -->
<!--   </tr> -->
<!--   </tr> -->
<!-- </tbody> -->
<!-- </table> -->
<!-- --- -->
<!-- ## Details -->
<!-- ### Introduction to R Part 1 {#intro1} -->
<!-- This workshop serves as a gentle introduction to programming with R. R basics will be covered, such as using the command line as a calculator, storing values in a variable, basic data structures (vector, matrix, list, data frame, etc.), and data types. We will show how to load basic data sets and access rows/columns, and summarize data with statistics and plots.  -->
<!-- #### Objectives -->
<!-- Participants will be able to: -->
<!-- - open the R command line interface (CLI) -->
<!-- - access the built-in help docs -->
<!-- - perform basic arithmetic using R -->
<!-- - store data in variables -->
<!-- - list the environment variables -->
<!-- - understand the basic data types in R -->
<!-- - convert between the different data types -->
<!-- - understand and use the basic data structures in R -->
<!-- - access rows, columns, and elements of matrices and data frames -->
<!-- - read data from a `csv` into a data frame -->
<!-- - fit simple linear models -->
<!-- - summarize data and linear models -->
<!-- - create simple plots of 1-D and 2-D data -->
<!-- ### Introduction to R Part 2 {#intro2} -->
<!-- This workshop introduces scripting in R, including control flow, writing functions, and installing/loading libraries (and the differences between CRAN, GitHub, and Bioconductor packages). We will teach you how to use RStudio.  We will go over the `-apply` family of functions, simple graphs, and good practices in R (style guide, organization, etc.). -->
<!-- #### Objectives -->
<!-- Participants will be able to: -->
<!-- - open up RStudio and identify key parts of the IDE -->
<!-- - create, open, and save R scripts -->
<!-- - run R scripts line by line or all at once -->
<!-- - understand and use the `-apply` family of functions -->
<!-- - use control flow to execute complex tasks -->
<!-- - write functions and understand when to use them -->
<!-- - install and load packages -->
<!-- - look up help for packages -->
<!-- - understand the differences between CRAN, Bioconductor, and GitHub -->
<!-- - understand what a style guide is and why it is important -->
<!-- - use good practices for writing and organizing code -->
<!-- ### Reproducible Data Science and tidyverse {#reproduce} -->
<!-- We will discuss the problems we commonly see and introduce what it means to conduct reproducible data science. During the workshop, we will go through an example of setting up a new project folder, loading "messy" data, organizing scripts, and creating basic Rmarkdown reports. In reference to the messy data, we will go over what "tidy" data is, and how to use `tidyverse` libraries. -->
<!-- #### Objectives -->
<!-- Participants will be able to: -->
<!-- - understand the principles of reproducible [data] science -->
<!-- - identify bad project management habits -->
<!-- - organize a new project folder -->
<!-- - write code that is human-readable -->
<!-- - identify helpful libraries for data wrangling -->
<!-- - understand and implement the principles of immutable data -->
<!-- - create reports and other media using Rmarkdown -->
<!-- - understand how to incorporate Github  -->
<!-- ### Producing Documents, Graphs, and Tables {#docs} -->
<!-- In this workshop, we will go over some principles of the visual display of quantitative data, and go in-depth with plotting in R with `base` and `ggplot2`. Additionally, since not all data is well represented in a graph, we will go over creating pretty tables using the `kable` and `kableExtra` packages. -->
<!-- #### Objectives -->
<!-- Participants will be able to -->
<!-- - understand the basic principles of the visual display of quantitative information -->
<!-- - produce clean and professional graphs using base `R` -->
<!-- - produce clean and professional graphs using `ggplot2` -->
<!-- - produce clean tables using `kable` and `kableExtra` -->
<!-- - create clean documents using `Rmarkdown` -->
</div>
</div>
<div id="acknowledgements" class="section level2">
<h2>Acknowledgements</h2>
<p>We thank Nevada INBRE (GM103440), UNR’s Graduate Student Association,
and UNR’s Graduate School for offering limited number of registration
waivers to students and postdoctoral fellows.</p>
<p>This bootcamp was also partially supported by grants from the
National Science Foundation (2203236 and 2141660).</p>
<p>Comics thanks to <a href="https://xkcd.com/"
class="uri">https://xkcd.com/</a>.</p>
</div>

<footer>
  <hr />
  Copyright &copy; 2023 Nevada INBRE. All rights reserved.
</footer>

<!-- Open external links in new tab -->
<script type="text/javascript" src="external_links.js"></script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
